internal/uvm/vsmb.go

```diff
@@ -69,7 +94,17 @@ func (uvm *UtilityVM) RemoveVSMB(ctx context.Context, hostPath string) error {
 
 	uvm.m.Lock()
 	defer uvm.m.Unlock()
-	share, err := uvm.findVSMBShare(ctx, hostPath)
+
+	st, err := os.Stat(hostPath)
+	if err != nil {
+		return err
+	}
+	m := uvm.vsmbDirShares
+	if !st.IsDir() {
+		m = uvm.vsmbFileShares
+		hostPath = filepath.Clean(filepath.Dir(hostPath))
```

Comment:
Same should we clean in all cases for key matches?

Reply:
Done